dioptas/model/util/NewFileWatcher.py

```diff
@@ -68,18 +68,20 @@ def __init__(self, path=None, file_types=None, activate=False):
         self.event_handler = PatternMatchingEventHandler(self.patterns)
         self.event_handler.on_created = self.on_file_created
 
-        self.active = False
+        self.active = activate
```

Comment:
Shouldn't this be kept as `self.active = False`?
Otherwise the call to `activate()` below won't do anything.

Reply:
but it will be false by default, because this is using the class initializer variable

Reply:
Yes, but if you pass`activate=True`, then `self.active` will be `True` when calling `self.activate()` below, which will not start the observer and the thread.

Reply:
However the `activate` parameter does not seems to be used.

Reply:
yes, i realized, you are right, it should be:

``` python
    self.active = False
    if activate:
        self.activate()
```